# 异常机制

cpp的异常 也是使用try ... catch ... 

但是和Java的异常处理机制又不太一样。

Java是catch exception，cpp 是catch 一个返回值，比如说打开文件发生异常，找不到文件返回 -1， 没有权限返回几，他是这样弄的。

所以，只能catch 一个int 。 然后 根据这个int 来决定如何处理。

或者说catch ... 这就代表着处理所有类型的异常。

# 异常接口声明

在函数后面声明异常

```cpp
int test()throw(int,float){}
```

![image-20240425232435821](https://raw.githubusercontent.com/heihei180/temp_file/file/picgo/image-20240425232435821.png)

![image-20240425232722022](https://raw.githubusercontent.com/heihei180/temp_file/file/picgo/image-20240425232722022.png)

## 异常类型和生命周期

![image-20240425232801666](https://raw.githubusercontent.com/heihei180/temp_file/file/picgo/image-20240425232801666.png)

函数一层一层执行，异常是逐层向上抛。

